server/api/include/rsFileChksum.hpp

```diff
@@ -1,13 +1,37 @@
 #ifndef RS_FILE_CHKSUM_HPP
 #define RS_FILE_CHKSUM_HPP
 
-#include "rodsConnect.h"
-#include "rodsDef.h"
+/// \file
+
 #include "fileChksum.h"
 
-int rsFileChksum( rsComm_t *rsComm, fileChksumInp_t *fileChksumInp, char **chksumStr );
-int _rsFileChksum( rsComm_t *rsComm, fileChksumInp_t *fileChksumInp, char **chksumStr );
-int remoteFileChksum( rsComm_t *rsComm, fileChksumInp_t *fileChksumInp, char **chksumStr, rodsServerHost_t *rodsServerHost );
-int fileChksum( rsComm_t *rsComm, char* objPath, char *fileName, char* rescHier, char* orig_chksum, char *chksumStr );
+struct RsComm;
+struct rodsServerHost;
+
+int rsFileChksum(RsComm *rsComm, fileChksumInp_t *fileChksumInp, char **chksumStr);
+
+int _rsFileChksum(RsComm *rsComm, fileChksumInp_t *fileChksumInp, char **chksumStr);
+
+int remoteFileChksum(RsComm *rsComm,
+                     fileChksumInp_t *fileChksumInp,
+                     char **chksumStr,
+                     rodsServerHost *rodsServerHost);
+
+[[deprecated("Use file_checksum.")]]
+int fileChksum(RsComm *rsComm,
+               char* objPath,
+               char *fileName,
+               char* rescHier,
+               char* orig_chksum,
+               char *chksumStr);
+
+int file_checksum(RsComm* _comm,
+                  const char* _logical_path,
+                  const char* _filename,
+                  const char* _resource_hierarchy,
+                  const char* _original_checksum,
+                  rodsLong_t _data_size,
+                  char* _calculated_checksum);
```

Comment:
Thoughts on the deprecation, name, etc.?

Reply:
looks good to me - was going to say make sure you get all of the server calls... but there were only two and they're handled in this PR.